solana/programs/ntt-quoter/src/processor/admin.rs

```diff
@@ -121,3 +121,73 @@ pub fn register_chain(ctx: Context<RegisterChain>, _args: RegisterChainArgs) ->
     ctx.accounts.registered_chain.base_price = u64::MAX;
     Ok(())
 }
+
+//TODO fix copypasta below
+
+#[derive(AnchorSerialize, AnchorDeserialize)]
+pub struct RegisterNttArgs {
+    pub ntt_program_id: Pubkey,
+    pub wormhole_transceiver_index: u8,
+    pub gas_cost: u32,
+}
+
+#[derive(Accounts)]
+#[instruction(args: RegisterNttArgs)]
+pub struct RegisterNtt<'info> {
+    #[account(
+        mut,
+        constraint = instance.is_authorized(&authority.key()) @ NttQuoterError::NotAuthorized
+    )]
+    pub authority: Signer<'info>,
+
+    pub instance: Account<'info, Instance>,
+
+    #[account(
+        init,
+        payer = authority,
+        space = 8 + RegisteredNtt::INIT_SPACE,
+        seeds = [RegisteredNtt::SEED_PREFIX, args.ntt_program_id.key().as_ref()],
+        bump,
+    )]
+    pub registered_ntt: Account<'info, RegisteredNtt>,
+
+    system_program: Program<'info, System>,
+}
+
+pub fn register_ntt(ctx: Context<RegisterNtt>, args: RegisterNttArgs) -> Result<()> {
+    ctx.accounts.registered_ntt.bump = ctx.bumps.registered_ntt;
+    ctx.accounts.registered_ntt.wormhole_transceiver_index = args.wormhole_transceiver_index;
```

Comment:
If this function is called twice, will it overwrite an existing index?
Maybe we could add a constraint for `wormhole_transceiver_index` to make sure that it is a new value. There's probably not too much risk but it would be good to prevent accidentally updating an existing entry.

I also think it could be good to constrain this to a value between 0-127 as the max number of transceivers is 128. (https://github.com/wormhole-foundation/example-native-token-transfers/issues/296)

Reply:
No such thing as calling it twice, because the `registered_ntt` account must be uninitalized (it's marked `init` in the context). If you want to change something, you'll have to deregister and re-register.

Regarding the max value of 128: If you register the wrong value, things will be broken anyway. I'd rather not duplicate what is effectively an implementation detail of the ntt program in the quoter, especially for something so marginal. Realistically, any value above 4 is almost certainly wrong.

Reply:
You're right, thanks I got tunnel vision on the possibility of overwriting the value and overlooked `init`.

Fair enough. I'm generally in favour of handling things explicitly but it's not worth blocking this PR